### PR TITLE
Smart Variables in hostgroup.

### PR DESCRIPTION
**Pull request by abenari**
1. Show smart variables in hostgroup view.
2. Added the GUI to override smart variables values in host group.

### CODE REVIEW

File: app/helpers/lookup_keys_helper.rb [hunk 1]
```diff
@@ -75,14 +75,33 @@ def overridable_lookup_keys klass, host
     klass.class_params.override.where(:environment_classes => {:environment_id => host.environment_id}) + klass.lookup_keys
   end
 
-  def key_with_diagnostic host, key
+  def key_with_diagnostic obj, key
+    if (obj.class.model_name=="host")
```
Comment (domcleal) [inline app/helpers/lookup_keys_helper.rb]: This is upper case "Host" on my local env, so it's loading the hostgroup helper on the host edit page.

File: app/views/hostgroups/_form.html.erb [hunk 2]
```diff
@@ -49,6 +49,13 @@
     </div>
 
     <div class="tab-pane" id="params">
+      <h6><%= _('Puppet classes Parameters') -%></h6>
+      <p/>
+      <%= render "puppetclasses/classes_parameters", { :obj => @hostgroup } %>
```
Comment (domcleal) [inline app/views/hostgroups/_form.html.erb]: There's some AJAX that refreshes this pane when you add a class to the host/hostgroup, but it's not working on the hostgroup.  Looks like the puppetclasses/parameters route and the load_puppet_class_parameters JS func are only handling hosts at the moment.

File: app/models/host_common.rb [hunk 3]
```diff
@@ -17,6 +17,27 @@ def self.included(base)
...
+          attr = attribute.dup
+          if attr.has_key? :id
+            lookup_value = lookup_values.find attr.delete(:id)
+            if lookup_value
+              mark_for_destruction = ActiveRecord::ConnectionAdapters::Column.value_to_boolean attr.delete(:_destroy)
+              lookup_value.attributes = attr
+              mark_for_destruction ? lookup_values.delete(lookup_value) : lookup_value.save!
+            end
+          elsif !ActiveRecord::ConnectionAdapters::Column.value_to_boolean attr.delete(:_destroy)
+            LookupValue.create(attr.merge(:match => lookup_value_match))
```
Comment (ohadlevy) [inline app/models/host_common.rb:37]: i think it should be
LookupValue.new(...)

File: app/views/hostgroups/_form.html.erb [hunk 4]
```diff
@@ -49,6 +49,13 @@
...
+      <h6><%= _('Hostgroup Parameters') -%></h6>
```
Comment (domcleal) [inline app/views/hostgroups/_form.html.erb]: Sorry, the style has changed in my recent i18n PRs, it should now be "Host group parameters".

File: app/models/host_common.rb (hunk 3 above)
Comment (abenari) [inline app/models/host_common.rb:37]: Tested both, it's working with create and nothing is saved if I replace to new.

**Comment by domcleal** [comment]
:+1: 

The other issue to link to in the commit message is #2220.

**Comment by ohadlevy** [comment]
merged and added the upstream ticket in the - thanks!

**Pull request closed without merging**